opencti-platform/opencti-graphql/src/modules/auth/auth-domain.ts

```diff
@@ -0,0 +1,139 @@
+import ejs from 'ejs';
+import { authenticator } from 'otplib';
+import { findById, getUserByEmail, userEditField } from '../../domain/user';
+import { AuthenticationFailure, UnsupportedError } from '../../config/errors';
+import { sendMail } from '../../database/smtp';
+import type { AuthContext } from '../../types/user';
+import type { AskSendOtpInput, Verify2faInput, User, VerifyOtpInput, ChangePasswordInput } from '../../generated/graphql';
+import { getEntityFromCache } from '../../database/cache';
+import type { BasicStoreSettings } from '../../types/settings';
+import { ENTITY_TYPE_SETTINGS } from '../../schema/internalObject';
+import { ADMIN_USER } from '../../../tests/utils/testQuery';
+import { OCTI_EMAIL_TEMPLATE } from '../../utils/emailTemplates/octiEmailTemplate';
+import { OTP_TTL, redisGetForgotPasswordOtp, redisSetForgotPasswordOtp } from '../../database/redis';
+import { publishUserAction } from '../../listener/UserActionListener';
+import { SYSTEM_USER } from '../../utils/access';
+import { killUserSessions } from '../../database/session';
+
+export const getUser = async (email: string): Promise<User> => {
```

Comment:
```suggestion
export const getLocalProviderUser = async (email: string): Promise<User> => {
```
If external users are not allowed, let's be explicit on the function name.